students/16th/team1/jinsookim/user/views.py

```diff
@@ -1,38 +1,65 @@
+        data         = json.loads(request.body)
         LEN_PASSWORD = 8
-    
-        phone_number = request['phone_number']
-        user_name    = request['user_name']
-        email        = request['email']
-        password     = request['password']
-   
-        if not phone_number and user_name or not email:
-            return JsonResponse({"message" :"KEY_ERROR"}, status = 400)
-
-        elif Users.objects.filter(user_name = user_name, phone_number = phone_number).exists() == True or Users.objects.filter(email = email).exists() == True:
-            return JsonResponse({"message" :"이미 회원가입이 되어 있습니다."}, status = 400)
-
-        elif not password:
-                return JsonResponse({"message" :"KEY_ERROR"}, status = 400)
-        elif int(password) < LEN_PASSWORD:
+        try:
+            phone_number = data['phone_number']
+            user_name    = data['user_name']
+            email        = data['email']
+            password     = data['password']
+
+            if Users.objects.filter(user_name = user_name, phone_number = phone_number).exists() == True or Users.objects.filter(email = email).exists() == True:
+                return JsonResponse({"message" :"이미 회원가입이 되어 있습니다."}, status = 400)
+        
+            elif len(password) < LEN_PASSWORD:
                 return JsonResponse({"message" :"비밀번호를 8자리 이상 입력해주세요."}, status = 400)
 
-        else:
             Users.objects.create(phone_number = phone_number,
-                             user_name    = user_name,
-                             email        = email,
-                             password     = password
-                             )
+                                user_name     = user_name,
+                                email         = email,
+                                password      = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())
+                                )
             return JsonResponse({"message" :"SUCCESS"}, status = 200)
 
-         
-        
+        except KeyError as e:
+            return JsonResponse({"message" : e.args[0]}, status = 400)
+    
         
         
+class Sign_in(View):
+    def post(self, request):
+        data  = json.loads(request.body)
+        q = Q()
+        try:
+            account  = data['account']
+            new_password = data['password']
+
+            if Users.objects.filter(Q(user_name=account) | Q(phone_number=account) | Q(email =account)).exists() == True and Users.objects.filter(password=bcrypt.checkpw(new_password.encode('utf-8'),password)).exists() == True:
+                return JsonResponse( {"message": "SUCCESS"}, status = 200)
+
+            return JsonResponse({"message": "INVALID_USER"}, status = 401) 
+            
+        except KeyError as e:
+            return JsonResponse({"message" : e.args[0]}, status = 400)
+
+
+def login_decorator(func):
+    def wrapper(self, request, *args, **kwargs):
+        data = json.loads(request.body)
+        user = data['user']
```

Comment:
```suggestion
email = data['email']
```
변수명은 어떤 의미인지 알 수 있게 지어주세요 

login_decorator 같은 경우에는 user view 에서만 사용하기 보다는 
여러곳에서 사용 할 수 있기 때문에, 절대경로에서 import 할 수 있게 해주는게 좋을 것 같에요